gulpfile.js

```diff
@@ -68,50 +72,98 @@ const {validator, validatorWebui} = require('./build-system/tasks/validator');
 const {vendorConfigs} = require('./build-system/tasks/vendor-configs');
 const {visualDiff} = require('./build-system/tasks/visual-diff');
 
-// Keep this list alphabetized.
-gulp.task('a4a', a4a);
-gulp.task('ava', ava);
-gulp.task('babel-plugin-tests', babelPluginTests);
-gulp.task('build', build);
-gulp.task('bundle-size', bundleSize);
-gulp.task('caches-json', cachesJson);
-gulp.task('check-exact-versions', checkExactVersions);
-gulp.task('check-links', checkLinks);
-gulp.task('check-owners', checkOwners);
-gulp.task('check-types', checkTypes);
-gulp.task('clean', clean);
-gulp.task('codecov-upload', codecovUpload);
-gulp.task('compile-jison', compileJison);
-gulp.task('create-golden-css', createGoldenCss);
-gulp.task('css', css);
-gulp.task('csvify-size', csvifySize);
-gulp.task('default', defaultTask);
-gulp.task('dep-check', depCheck);
-gulp.task('dev-dashboard-tests', devDashboardTests);
-gulp.task('dist', dist);
-gulp.task('e2e', e2e);
-gulp.task('firebase', firebase);
-gulp.task('generate-vendor-jsons', generateVendorJsons);
-gulp.task('get-zindex', getZindex);
-gulp.task('integration', integration);
-gulp.task('lint', lint);
-gulp.task('make-extension', makeExtension);
-gulp.task('nailgun-start', nailgunStart);
-gulp.task('nailgun-stop', nailgunStop);
-gulp.task('performance', performance);
-gulp.task('pr-check', prCheck);
-gulp.task('prepend-global', prependGlobal);
-gulp.task('presubmit', presubmit);
-gulp.task('prettify', prettify);
-gulp.task('process-3p-github-pr', process3pGithubPr);
-gulp.task('process-github-issues', processGithubIssues);
-gulp.task('serve', serve);
-gulp.task('size', size);
-gulp.task('todos:find-closed', todosFindClosed);
-gulp.task('unit', unit);
-gulp.task('update-packages', updatePackages);
-gulp.task('validator', validator);
-gulp.task('validator-webui', validatorWebui);
-gulp.task('vendor-configs', vendorConfigs);
-gulp.task('visual-diff', visualDiff);
-gulp.task('watch', watch);
+/**
+ * Creates a gulp task using the given name and task function.
+ *
+ * @param {string} name
+ * @param {function} taskFunc
+ */
+function createTask(name, taskFunc) {
+  checkFlags(name, taskFunc);
```

Comment:
Consider adding `--undefok` 😂
(you might think I'm kidding but this could help during migrations, i.e. flag renames...)

Reply:
I read the documentation for `undefok`.

- If the goal is to temporarily allow more than one alternative for a given flag, there's already a way to do so (by adding both alternatives to `fooTask.flags`).
- If the goal is add a workaround to use invalid flags, that's the opposite of what this PR is trying to do :)